src/SafeFile.cpp

```diff
@@ -57,6 +59,9 @@ bool SafeFile::close()
 {
     if (!f)
         return false;
+#ifdef ARCH_NRF52
+    f.truncate();
+#endif
 
     spiLock->lock();
```

Comment:
Should this lock happen before the truncate?

Reply:
Indeed. Pushed the change